src/wallet/wallet.cpp

```diff
@@ -245,7 +245,7 @@ bool CWallet::Unlock(const SecureString& strWalletPassphrase, bool anonymizeOnly
 {
     SecureString strWalletPassphraseFinal;
 
-    if (!IsLocked())
+    if (!IsLocked() && !fWalletUnlockAnonymizeOnly) // was already fully unlocked, not only for anonymization
```

Comment:
I think yes and that's the idea - if you have wallet unlocked for mixing only you must go through the same unlock procedure when you want to unlock it for "normal" sending of funds etc i.e. generate encryption key from passphrase and try to decrypt wallet keys again.